oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/core/bearer/OAuth2AccessTokenAuthority.java

```diff
@@ -1,85 +0,0 @@
-/*
- * Copyright 2002-2018 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.springframework.security.oauth2.core.bearer;
```

Comment:
What are your thoughts about where attributes should be stored? It seems in other code that the attributes are stored as part of the authority.

I actually liked this because it gave the authoritiesMapper a bit more control, should we not have the right hook a dev needs to appropriately transform the attributes. (I got this idea from your OAuth2 documentation on authorities mappers).

Reply:
> What are your thoughts about where attributes should be stored?

It's natural that attributes are composed within a `Jwt`, `IdToken`, or `IntrospectedAccessToken` (future).

> It seems in other code that the attributes are stored as part of the authority.

The `oauth2Login()`  feature which is implemented this way is different from my viewpoint. I don't feel the same approach applies to Resource Server. Let's definitely have a chat and figure out our approach with Authorities mapping for Resource Server.

Reply:
See the comments about the authorities mapper in this same PR for some additional discussion surrounding this point.